File: Data Structures & Algorithms/invert-a-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return root;
        }
        Deque<TreeNode> s = new ArrayDeque<>();
        s.add(root);
        while(!s.isEmpty()){
            var n = s.pop();
            var temp = n.left;
            n.left = n.right;
            n.right = temp;
            if(n.left!=null) s.push(n.left);
            if(n.right!=null) s.push(n.right);
        }
        return root;
    }
}

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

type Entry struct {
	node *TreeNode
	top int
}

func goodNodes(root *TreeNode) int {
    if root == nil {
		return 0
	}
	ans := 0
	q := make([]Entry,0,100)
	q = append(q, Entry{root,-101})
	for len(q) > 0 {
		cur := q[len(q)-1]
		q = q[:len(q)-1]
		if cur.node.Val >= cur.top {
			ans++
			cur.top = cur.node.Val
		}

		if cur.node.Left != nil {
			q = append(q, Entry{cur.node.Left,cur.top})
		}
		
		if cur.node.Right != nil {
			q = append(q, Entry{cur.node.Right,cur.top})
		}
	}
	return ans
}
